momento-sdk/src/main/java/momento/sdk/messages/CacheGetResponse.java

```diff
@@ -1,109 +1,80 @@
-   * @return The result of Cache Get Operation
-   */
-  public CacheGetStatus status() {
-    return status;
-  }
+    public Hit(ByteString value) {
+      this.value = value;
+    }
 
-  /**
-   * Value stored in the cache as a byte array.
-   *
-   * @return Value stored for the given key. {@link Optional#empty()} if the lookup resulted in a
-   *     cache miss.
-   */
-  public Optional<byte[]> byteArray() {
-    if (status != CacheGetStatus.HIT) {
-      return Optional.empty();
+    /**
+     * Gets the retrieved value as a byte array.
+     *
+     * @return the value.
+     */
+    public byte[] valueByteArray() {
+      return value.toByteArray();
     }
-    return Optional.ofNullable(body.toByteArray());
-  }
 
-  /**
-   * Value stored in the cache as a {@link ByteBuffer}.
-   *
-   * @return Value stored for the given key. {@link Optional#empty()} if the lookup resulted in a
-   *     cache miss.
-   */
-  public Optional<ByteBuffer> byteBuffer() {
-    if (status != CacheGetStatus.HIT) {
-      return Optional.empty();
+    /**
+     * Gets the retrieved value as a {@link ByteBuffer}.
+     *
+     * @return the value.
+     */
+    public ByteBuffer valueByteBuffer() {
+      return value.asReadOnlyByteBuffer();
     }
-    return Optional.ofNullable(body.asReadOnlyByteBuffer());
-  }
 
-  /**
-   * Value stored in the cache as a UTF-8 {@link String}
-   *
-   * @return Value stored for the given key. {@link Optional#empty()} if the lookup resulted in a
-   *     cache miss.
-   */
-  public Optional<String> string() {
-    return string(StandardCharsets.UTF_8);
-  }
+    /**
+     * Gets the retrieved value as a UTF-8 {@link String}
+     *
+     * @return the value.
+     */
+    public String valueString() {
+      return valueString(StandardCharsets.UTF_8);
+    }
 
-  /**
-   * Value stored in the cache as {@link String}.
-   *
-   * @param charset to express the bytes as String.
-   * @return Value stored for the given key. {@link Optional#empty()} if the lookup resulted in a
-   *     cache miss.
-   */
-  public Optional<String> string(Charset charset) {
-    if (status != CacheGetStatus.HIT) {
-      return Optional.empty();
+    /**
+     * Gets the retrieved value as a {@link String}.
+     *
+     * @param charset the string encoding to use.
+     * @return the value.
+     */
+    public String valueString(Charset charset) {
+      return value.toString(charset);
     }
-    return Optional.ofNullable(body.toString(charset));
-  }
 
-  /**
-   * Value as an {@link InputStream}
-   *
-   * @return Value stored for the given key. {@link Optional#empty()} if the lookup resulted in a
-   *     cache miss.
-   */
-  public Optional<InputStream> inputStream() {
-    if (status != CacheGetStatus.HIT) {
-      return Optional.empty();
+    /**
+     * Gets the retrieved value as an {@link InputStream}.
+     *
+     * @return the value.
+     */
+    public InputStream valueInputStream() {
+      return value.newInput();
```

Comment:
Same with this one, ie do we want the minimum set of value accessors to: utf-8 string, other encoded string, byte array, byte buffer, input stream? If not relevant for this PR, ignore this :-)

Reply:
I also don't know if we really need all of those methods in the response. I just moved over the existing ones. @cprice404 what are your feelings on having a batteries included response like this vs one that just returns a byte[] and string?

Reply:
huge +1 from me on trimming these down.  we need string, and we need one bytes accessor and i'd stop there.

the question is, for the bytes one, which type should we expose?  tbh i'm tempted to choose `inpustream`.

@kvcache  any thoughts?

Reply:
Our GetResponse (and all payload objects) are fully materialized: I vote we should give users byte[] and String (and String only because we have taken on the utf-8 symmetric functionality).

Hiding their `byte[]` behind an InputStream will only annoy someone who is using bytes APIs (hello protocol buffers), and it won't matter at all to someone who is using nio streams (trivial to wrap a byte[] in your stream type of choice).

We could possibly in the future support huge items for which we would vend batches of bytes at a time. In that possible future, I think returning an InputStream would be the most truthful and friendly type available.

Reply:
Updated.